ui/notifications.ts

```diff
@@ -0,0 +1,180 @@
+</div>`;
+
+export class NotificationManager {
+
+    private static NOTIFICATION_KEY = 'service_notifications';
+
+    private notifications: Notification[] = [];
+
+    getNotifications(): Notification[] {
+        return this.notifications;
+    }
+
+    dismissNotification(id: number) {
+        const index = this.notifications.findIndex(v => v.id === id);
+        if (index > -1) {
+            this.notifications.splice(index, 1);
+        }
+        this.toCache(this.notifications);
+        console.log(this.notifications);
+    }
+
+    mergeToCache(notifications: Notification[]) {
+        const data = this.fromCache().concat(notifications);
+        this.toCache(data);
+        localStorage.setItem(NotificationManager.NOTIFICATION_KEY, JSON.stringify(data));
+        this.notifications = data;
+    }
+
+    clearAll() {
+        this.notifications = [];
+        this.toCache(this.notifications);
+    }
+
+    getNextId(): number {
+        const ids = this.notifications.map(value => value.id)
+            .sort((a, b) => b - a);
+
+        let nextId: number;
+        if (ids.length > 0) {
+            nextId = ids[0] + 1;
+        } else {
+            nextId = 1;
+        }
+        return nextId;
+    }
+
+    private toCache(notifications: Notification[]) {
+        localStorage.setItem(NotificationManager.NOTIFICATION_KEY, JSON.stringify(notifications));
+    }
+
+    private fromCache(): Notification[] {
+        const stored = localStorage.getItem(NotificationManager.NOTIFICATION_KEY);
+        let cached: Notification[] = [];
+        if (stored) {
+            cached = JSON.parse(stored);
+        }
+        return cached;
+    }
+}
+
+export const notificationManager = new NotificationManager();
+
+function updateNotificationMessages(notifications: Notification[], elements: JQuery<HTMLElement>) {
+    for (const notification of notifications) {
+        const element = elements.filter(`#notification-${notification.id}`);
+        if (element.html()) {
+            continue;
+        }
+        $('.notification-area').prepend(singleNotification(notification));
+    }
+
+}
+export function updateNotifications() {
+    const notifications = notificationManager.getNotifications();
+    const badge = $('#notification-badge');
+    const messages = $('#notification-messages');
+
+    badge.html(notificationBadge(notifications));
+
+    const elements = $('.notification');
+    if (elements.length === 0) {
+        messages.html(notificationMessages(notifications));
+    } else {
+        updateNotificationMessages(notifications, elements);
+    }
+
+}
+export function setupNotificationHandlers() {
+    const $notification = $('.notification');
+
+    const $clear = $('#notifications-clear-all');
+    $notification.off('click');
+
+    $clear.off('click');
+
+    $notification.on('click', function (event: JQuery.Event) {
+        event.preventDefault();
+        $(event.target).closest('.notification').remove();
```

Comment:
Another thing that I just thought about is that with these notifications the old notifications (the bell thing should go away). I don't think anything is going to be appearing there anymore, right?

Reply:
I can use the bell then for the new notifications and remove the old one?

Reply:
I really have no idea. I need to check when that part is updated, and maybe if there are any messages, integrate with the new system.

Reply:
I think it used to be that I checked the logs for warnings and put them there. But this was a bad idea and I stopped it. I think it should just go away.